databricks/koalas/indexes/base.py

```diff
@@ -507,6 +507,29 @@ def to_numpy(self, dtype=None, copy=False) -> np.ndarray:
             result = result.copy()
         return result
 
+    def map(
+        self,
+        mapper: Union[dict, Callable[[Any], Any], pd.Series],
+        return_type: ks.typedef.Dtype = str,
```

Comment:
pandas doesn't accept `return_type` argument. https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Index.map.html

Reply:
Btw, shall we import `Dtype` in the header if we still need it?

Reply:
Also, `str` is not `Dtype`.

https://github.com/databricks/koalas/blob/f845001c444b1aebf6ee78245ed4474e093c62fd/databricks/koalas/typedef/typehints.py#L75

Reply:
I'll fix, I added the return_type parameter to be leveraged by the `pandas_udf` that is used to execute the transformation. Will think of a way to make this discoverable at run time removing the `return_type` parameter.